plugins/filter_aws/aws.c

```diff
@@ -370,6 +400,208 @@ static int get_vpc_metadata(struct flb_filter_aws *ctx)
+        return -1;
+    }
+
+    /* go over the response and initialize tag_keys values */
+    /* code below finds two indices which define tag key and copies them to ctx */
+    while (tag_end <= len) {
+        /* replace \n with \0 to 'clearly' separate tag key strings */
+        if (tags_list[tag_end] == '\n') {
+            tags_list[tag_end] = '\0';
+        }
+        if ((tags_list[tag_end] == '\0' || tag_end == len) && (tag_start < tag_end)) {
+            /* length of tag key characters is the difference between start and end */
+            /* for instance, if tag name is 'Name\0...', the corresponding values are */
+            /*   tag_start = 0, points to 'N' */
+            /*   tag_end = 4, points to '\0' just after 'e' */
+            /*   f.e.: 4 - 0 = 4, which is equal to len("Name") */
+            tag_key_len = tag_end - tag_start;
+            ctx->tag_keys_len[tag_index] = tag_key_len;
+
+            /* allocate new memory for the tag key value */
+            /* + 1, because we need one more character for \0 */
+            ctx->tag_keys[tag_index] = flb_sds_create_size(tag_key_len + 1);
+            if (!ctx->tag_keys[tag_index]) {
+                flb_errno();
+                flb_sds_destroy(tags_list);
+                return -2;
+            }
+
+            /* tag_key points to the first character of tag key as char* */
+            tag_key = tags_list + tag_start;
+            strcpy(ctx->tag_keys[tag_index], tag_key);
+
+            flb_plg_debug(ctx->ins, "tag found: %s (len=%d)", ctx->tag_keys[tag_index],
+                          tag_key_len);
+
+            tag_index++;
+            tag_start = tag_end + 1;
+        }
+        tag_end++;
+    }
+
+    flb_sds_destroy(tags_list);
+
+    return ret;
+}
+
+/* Get EC2 instance tag values from /latest/meta-data/tags/instance/{tag_key}.
+ * Initializes ctx->tag_values and ctx->tag_values_len.
+ */
+static int get_ec2_tag_values(struct flb_filter_aws *ctx)
+{
+    int ret;
+    size_t i;
+    flb_sds_t tag_value = NULL;
+    size_t tag_value_len = 0;
+    size_t tag_value_path_len;
+    flb_sds_t tag_value_path;
+
+    /* initialize array for the tag values */
+    ctx->tag_values = flb_calloc(ctx->tags_count, sizeof(flb_sds_t*));
+    if (!ctx->tag_values) {
+        flb_errno();
+        return -1;
+    }
+    ctx->tag_values_len = flb_calloc(ctx->tags_count, sizeof(size_t*));
+    if (!ctx->tag_values_len) {
+        flb_errno();
+        return -1;
+    }
+
+    for (i = 0; i < ctx->tags_count; i++) {
+        /* fetch tag value using path: /latest/meta-data/tags/instance/{tag_name} */
+        tag_value_path_len = ctx->tag_keys_len[i] + 1 +
+                             strlen(FLB_FILTER_AWS_IMDS_INSTANCE_TAG);
+        tag_value_path = flb_sds_create_size(tag_value_path_len);
+        if (!tag_value_path) {
+            flb_errno();
+            return -1;
+        }
+        tag_value_path = flb_sds_printf(&tag_value_path, "%s/%s",
```

Comment:
check flb_sds_printf failure

Reply:
I've tried to find an example in other parts of the codebase, but it seems failures from `flb_sds_printf` are rarely checked. Please let me know if what I've done is correct. https://github.com/fluent/fluent-bit/pull/6065/commits/3366de590b9ce8094589bd6eaf48b9c005e7f926

Reply:
> but it seems failures from flb_sds_printf are rarely checked.

That's a bug.... that's why I recently had to do this: https://github.com/fluent/fluent-bit/pull/5930

The idiomatic Fluent Bit way to do it is:

```
tmp = flb_sds_printf(blah...
if (!tmp) {
    flb_errno();
    // more error handling if needed
}
final_val = tmp
```